File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{
    var answer = [];
    let idx = 0;
    while(idx != arr.length){
        // 일단 숫자 하나 저장
        // 뒤랑 다르면 arr[idx] != arr[idx + 1]
        // idx += 1;
        if(arr[idx] != arr[idx+1]){
            answer.push(arr[idx]);
        }
        idx += 1;
    }
    
    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer> stack = new Stack<>();
        for(int a : arr){
            while(!stack.isEmpty() && stack.peek() == a){
                stack.pop();
            }
            stack.push(a);
        }       
        
        int[] answer = new int[stack.size()];
        for(int i = stack.size() - 1; i >= 0; i--){
            answer[i] = stack.pop();
        }
        return answer;
    }
}